jboss-container-images/redhat-sso-7-openshift-image
# [KEYCLOAK-15633] Add support for "${prefix}_CONNECTION_PROPERTY_*" for non-XA datasources too

<hr/>

<b>Note (for myself):</b> Once the change is accepted / merged, the value of `POSTGRESQL_SOURCE_REPOSITORY_URL` variable from `sso75-x509-postgresql-persistent.json` template needs to be updated to point to the main jboss-container-images RH-SSO images repository (rather than my fork of it).

<hr/>

<u>TL;DR: The short summary of the (intentionally kept separate) commits is as follows:</u>

1. Add `s2i` build definition to enable SSL/TLS support for RHSCL RHEL-7 PostgreSQL server container image,
2. Start PostgreSQL server with SSL/TLS by default when launched from `"sso75-x509-postgresql-persistent.json"` template
3. Add support for injection of connection properties specified via `${prefix}_CONNECTION_PROPERTY_*` environment variables passed to RH-SSO container image. These connection properties would be added to the definition of each non-XA datasource, that would be configured for the image

<hr/>

To test:

1. `oc new-project postgresql-ssl-demo`
2. `oc import-image <image_location>:<image_tag> --insecure=true --confirm`
3. `oc get is` and save the image stream name & image stream tag from it
4. Modify `from:` element / field of the RH-SSO DeploymentConfig of the `"sso75-x509-postgresql-persistent.json"` from this PR to contain the image stream name & image stream tag obtained in previous step. Example:
```
$ diff /tmp/sso75-x509-postgresql-persistent.json.orig /tmp/sso75-x509-postgresql-persistent.json
395,396c395
<                                 "namespace": "${IMAGE_STREAM_NAMESPACE}",
<                                 "name": "sso75-openshift-rhel8:7.5"
---
>                                 "name": "rh-sso-7-sso75-openshift-rhel8:rh-sso-7.5-rhel-8-containers-candidate-50841-20211026141326"

```
IOW the `"namespace"` field was removed, the image to be searched in the current namespace (instead of the default `openshift` one). And image stream name & image stream tag were updated to match the actual values.
5. Create instance of the modified template
    ``` oc create -f /tmp/sso75-x509-postgresql-persistent.json```
6. (Optional) Verify it got created correctly:
  ```
  $ oc get templates
NAME                               DESCRIPTION                                                                        PARAMETERS     OBJECTS
sso75-x509-postgresql-persistent   An example application based on RH-SSO 7.5 on OpenJDK image. For more informa...   24 (9 blank)   9
  ```
7.  Deploy the RH-SSO pod using the modified template
  ```
  $ oc new-app --template=sso75-x509-postgresql-persistent
  ```
8.  Modify the `sso` deployment to contain various `DB_CONNECTION_PROPERTY_*` (& possibly also combined with `DB_XA_CONNECTION_PROPERTY_*` env vars) and confirm they were properly set on the underlying `<datasource ` definition (and / or `<xa-datasource` definition) respectively.

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`
